roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3511 @@
+}
+
+std::vector<std::vector<RooArgList *>> const&
+RooLagrangianMorphFunc::Config::getDiagrams() const {
+  return this->_configDiagrams;
+}
+
+RooArgList RooLagrangianMorphFunc::Config::getCouplings() const {
+  return this->_couplings;
+}
+
+RooArgList RooLagrangianMorphFunc::Config::getProdCouplings() const {
+  return this->_prodCouplings;
+}
+
+RooArgList RooLagrangianMorphFunc::Config::getDecCouplings() const {
+  return this->_decCouplings;
+}
+
+RooArgList RooLagrangianMorphFunc::Config::getFolders() const {
+  return this->_folderlist;
+}
+
+Bool_t RooLagrangianMorphFunc::Config::IsAllowNegativeYields() const {
+  return this->_allowNegativeYields;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// print all the parameters and their values in the given sample to the console
+
+void RooLagrangianMorphFunc::printParameters(const char *samplename) const {
+  for (const auto &param : this->_config.getParamCards().at(samplename)) {
+    if (this->hasParameter(param.first.c_str())) {
+      std::cout << param.first << " = " << param.second;
+      if (this->isParameterConstant(param.first.c_str()))
+        std::cout << " (const)";
+      std::cout << std::endl;
+    }
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// print all the known samples to the console
+
+void RooLagrangianMorphFunc::printSamples() const {
+  // print all the known samples to the console
+  for (auto folder : this->_config.getFolderNames()) {
+    std::cout << folder << std::endl;
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// print the current phyiscs values
+
+void RooLagrangianMorphFunc::printPhysics() const {
+  for (const auto &sample : this->_sampleMap) {
+    RooAbsArg *phys = this->_physics.at(sample.second);
+    if (!phys)
+      continue;
+    phys->Print();
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+///  return the number of samples in this morphing function
+
+int RooLagrangianMorphFunc::Config::nSamples() const {
+  return this->_folderNames.size();
+}
+
+int RooLagrangianMorphFunc::nSamples() const {
+  return this->_config.getFolderNames().size();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// protected constructor with proper arguments
+
+RooLagrangianMorphFunc::RooLagrangianMorphFunc(const char *name,
+                                               const char *title,
+                                               const Config &config)
+    : RooAbsReal(name, title), _cacheMgr(this, 10, kTRUE, kTRUE),
+      _operators("operators", "set of operators", this, kTRUE, kFALSE),
+      _observables("observables", "morphing observables", this, kTRUE, kFALSE),
+      _binWidths("binWidths", "set of binWidth objects", this, kTRUE, kFALSE),
+      _config(config), _curNormSet(0) {
+  this->printAuthors();
```

Comment:
```suggestion
```
I don't think printing the authors every time a `RooLagrangianMorphFunc` gets created is necessary.